datadog/writer.go

```diff
@@ -13,75 +13,90 @@ import (
 	"github.com/cultureamp/glamplify/log"
 )
 
-// writerConfig for setting initial values for Monitor Writer
-type writerConfig struct {
-	// license is your Data Dog API key.
+type DDWriter interface {
+	WriteFields(sev int, system log.Fields, fields ...log.Fields)
+	WaitAll()
+}
+
+// DDFieldWriter sends logging output to Data Dog
+type DDFieldWriter struct {
+	// ApiKey for Data Dog API key.
 	//
 	// https://app.datadoghq.com/account/settings#api
-	apiKey string
+	ApiKey string
 
-	// URL: https://http-intake.logs.datadoghq.com/v1/input  (default)
-	endpoint string
+	// Endpoint URL: https://http-intake.logs.datadoghq.com/v1/input  (default)
+	Endpoint string
 
-	// timeout
-	timeout time.Duration
+	// Timeout on HTTP requests
+	Timeout time.Duration
 
-	omitempty bool
-}
+	// Omitempty will remove empty fields before sending
+	Omitempty bool
 
-// FieldWriter sends logging output to Data Dog
-type FieldWriter struct {
-	mutex  sync.Mutex
-	config writerConfig
+	// WaitGroup can optionally be to a valid wait group, and the writer will signal when it sends and completes
+	// so clients can
+	WaitGroup *sync.WaitGroup
 }
 
 // NewDataDogWriter creates a new FieldWriter. The optional configure func lets you set values on the underlying  writer.
-func NewDataDogWriter(configure ...func(*writerConfig)) *FieldWriter { // https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
-	conf := writerConfig{
-		apiKey:    os.Getenv("DD_CLIENT_API_KEY"),
-		endpoint:  helper.GetEnvString("DATA_DOG_LOG_ENDPOINT", "https://http-intake.logs.datadoghq.com/v1/input"),
-		timeout:   time.Second * time.Duration(helper.GetEnvInt("DATA_DOG_TIMEOUT", 5)),
-		omitempty: helper.GetEnvBool(log.OmitEmpty, false),
+func NewDataDogWriter(configure ...func(*DDFieldWriter)) DDWriter { // https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
+	writer := &DDFieldWriter{
+		ApiKey:    os.Getenv("DD_CLIENT_API_KEY"),
+		Endpoint:  helper.GetEnvString("DATA_DOG_LOG_ENDPOINT", "https://http-intake.logs.datadoghq.com/v1/input"),
+		Timeout:   time.Second * time.Duration(helper.GetEnvInt("DATA_DOG_TIMEOUT", 5)),
+		Omitempty: helper.GetEnvBool(log.OmitEmpty, false),
```

Comment:
```suggestion
		OmitEmpty: helper.GetEnvBool(log.OmitEmpty, false),
```
I suggest Pascal case to align with Go's norms

Reply:
opps - missed this. Fixing now...